backend/src/bork-protocol/services/image/image-generation.ts

```diff
@@ -0,0 +1,193 @@
+
+    const model = options.model || 'dall-e-3';
+    const defaultSize = model === 'dall-e-3' ? '1792x1024' : '1024x1024';
+
+    const requestBody = {
+      model,
+      prompt,
+      n: model === 'dall-e-3' ? 1 : options.numberOfImages || 1,
+      size: options.size || defaultSize,
+      quality: options.quality || 'standard',
+      style: options.style || 'vivid',
+      response_format: options.responseFormat || 'url',
+      user: options.user,
+    };
+
+    const response = await fetch(`${endpoint}/images/generations`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${this.runtime.getSetting('OPENAI_API_KEY')}`,
+      },
+      body: JSON.stringify(requestBody),
+    });
+
+    if (!response.ok) {
+      await handleApiError(response, 'OpenAI');
+    }
+
+    const data = (await response.json()) as {
+      created: number;
+      data: Array<{ url: string }>;
+    };
+
+    return {
+      imageUrls: data.data.map((item) => item.url),
+      metadata: {
+        created: data.created,
+        model: requestBody.model,
+        ...options,
+      },
+    };
+  }
+}
+
+export class ImageGenerationService extends Service {
+  static serviceType: ServiceType = ServiceType.IMAGE_DESCRIPTION;
+
+  private initialized = false;
+  private runtime: IAgentRuntime | null = null;
+  private provider: ImageProvider | null = null;
+
+  async initialize(runtime: IAgentRuntime): Promise<void> {
+    elizaLogger.log('Initializing ImageGenerationService');
+    this.runtime = runtime;
+  }
+
+  private async initializeProvider(): Promise<boolean> {
+    if (!this.runtime) {
+      throw new Error('Runtime is required for image generation');
+    }
+
+    this.provider = new OpenAIImageProvider(this.runtime);
+    elizaLogger.debug('Using OpenAI for image generation model');
+
+    try {
+      await this.provider.initialize();
+      return true;
+    } catch {
+      elizaLogger.error(
+        'Failed to initialize the OpenAI image generation provider',
+      );
+      return false;
+    }
+  }
+
+  async generateImage(
+    prompt: string,
+    options?: ImageGenerationOptions,
+  ): Promise<ImageGenerationResult> {
+    if (!this.initialized) {
+      this.initialized = await this.initializeProvider();
+    }
+
+    if (!this.initialized || !this.provider) {
+      throw new Error('Image generation service not properly initialized');
+    }
+
+    try {
+      return await this.provider.generateImage(prompt, options);
+    } catch (err) {
+      elizaLogger.error('Error in generateImage:', err);
+      throw err;
+    }
+  }
+
+  async saveGeneratedImage(
+    imageUrl: string,
+    outputPath: string,
+  ): Promise<string> {
+    try {
+      const response = await fetch(imageUrl);
+      if (!response.ok) {
+        throw new Error(`Failed to download image: ${response.statusText}`);
+      }
+
+      const buffer = Buffer.from(await response.arrayBuffer());
```

Comment:
IMO should save it in supabase, but I dont think thats necessary for now. But its a nice addon!